### ISSUE TITLE
Temperature seems to be in fahrenheit, even though the unit is set to celcius

### ISSUE THREAD

**Issue by dkgof** [sanjay900/igrill#4]
The sensor seems to report the value of probe as fahrenheit, even though the unit is set to celcius.

At least for my igrill mini.

Changing this function in igrill.py, to convert to actual celcius value, seems to work:

```
def update_temp_sensor(self, payload, name):
        temp = payload[0] + (payload[1] * 256)
        if float(temp) != 63536.0:
            temp = float(temp)
            temp = round((temp - 32.0) * 5.0/9.0, 1)
        else:
            temp = 0
        temp_unit = SensorLibrary.TEMPERATURE__CELSIUS
        self.update_predefined_sensor(temp_unit, temp, name)
        self.update_listeners()```

**Comment by sanjay900** [comment]
Thats odd because my igrill v2 is defo reporting in celcius, i think you can set the temperature unit on the igrill itself?
Is it possible that when you use the offical app to set a temperature unit, it actually changes what is reported by the igrill?
I will see if i can find some way to read out the  current unit from the igrill itself.

**Comment by dkgof** [comment]
I dont see any way to switch on the device itself, only the on/off button on the top, and the reset button inside the battery compartment. I wonder how it is supposed to work.

**Comment by sanjay900** [comment]
What unit does it show when you use the phone app? Maybe this is just a weird limitation of the mini, since i have a v2 that doesnt have this issue

**Comment by dkgof** [comment]
I can choose between C and F in the app.

**Comment by dkgof** [comment]
But maybee if i install the app and change it, it changes on the device?

To be fair i have not used the app in ages. since i switched to a mqtt readout a few years ago, and then recently switched to your component.

I will try and poke around inside the app.

**Comment by dkgof** [comment]
Weird, the app just shows C now, and no possibility to change it. I am extremely confused now, because i am 100% sure that the values i was getting in HA was in F. Now it seems it will not connect anymore though so i will have to fiddle with it.

**Comment by dkgof** [comment]
I got it to reconnect by going into bluetoothctl and removing the pairing with the device, then it repaired itself and is now showing the temp correctly in C, without my change. So i am guessing that the app set something inside the device, so its now reporting in C. And since i did a reset of the device when first trying to pair it with your component, i probabely reset it back to F?

Anyways, it seems it was fixed by pairing with the app, on a device with region set to somewhere where C is the default.

Sorry for the trouble :)
